File: src/main/java/com/weeth/domain/user/presentation/UserResponseCode.java
package com.weeth.domain.user.presentation;

import com.weeth.global.common.response.ResponseCodeInterface;
import lombok.Getter;
import org.springframework.http.HttpStatus;

@Getter
public enum UserResponseCode implements ResponseCodeInterface {
    // UserAdminController 관련
    USER_FIND_ALL_SUCCESS(1800, HttpStatus.OK, "모든 회원 정보를 성공적으로 조회했습니다."),
    USER_DETAILS_SUCCESS(1801, HttpStatus.OK, "특정 회원의 상세 정보를 성공적으로 조회했습니다."),
    USER_ACCEPT_SUCCESS(1802, HttpStatus.OK, "회원 가입 승인이 성공적으로 처리되었습니다."),
    USER_BAN_SUCCESS(1803, HttpStatus.OK, "회원이 성공적으로 차단되었습니다."),
    USER_ROLE_UPDATE_SUCCESS(1804, HttpStatus.OK, "회원의 역할이 성공적으로 수정되었습니다."),
    USER_APPLY_OB_SUCCESS(1805, HttpStatus.OK, "OB 신청이 성공적으로 처리되었습니다."),
    USER_PASSWORD_RESET_SUCCESS(1806, HttpStatus.OK, "비밀번호가 성공적으로 초기화되었습니다."),
    // UserController 관련
    USER_APPLY_SUCCESS(1807, HttpStatus.OK, "회원 가입 신청이 성공적으로 처리되었습니다."),
    USER_EMAIL_CHECK_SUCCESS(1808, HttpStatus.OK, "이메일 중복 검사가 성공적으로 처리되었습니다."),
    USER_FIND_BY_ID_SUCCESS(1809, HttpStatus.OK, "회원 정보가 성공적으로 조회되었습니다."),
    USER_UPDATE_SUCCESS(1810, HttpStatus.OK, "회원 정보가 성공적으로 수정되었습니다."),
    USER_LEAVE_SUCCESS(1811, HttpStatus.OK, "회원 탈퇴가 성공적으로 처리되었습니다."),
    SOCIAL_LOGIN_SUCCESS(1812, HttpStatus.OK, "소셜 로그인에 성공했습니다."),
    SOCIAL_REGISTER_SUCCESS(1813, HttpStatus.OK, "소셜 회원가입에 성공했습니다."),
    SOCIAL_AUTH_SUCCESS(1814, HttpStatus.OK, "소셜 인증에 성공했습니다."),
    SOCIAL_INTEGRATE_SUCCESS(1815, HttpStatus.OK, "소셜 로그인 연동에 성공했습니다."),
    JWT_REFRESH_SUCCESS(1816, HttpStatus.OK, "토큰 재발급에 성공했습니다."),

    // CardinalController 관련
    CARDINAL_FIND_ALL_SUCCESS(1817, HttpStatus.OK, "전체 기수 조회에 성공했습니다."),
    CARDINAL_SAVE_SUCCESS(1818, HttpStatus.OK, "기수 저장에 성공했습니다."),
    CARDINAL_UPDATE_SUCCESS(1819, HttpStatus.OK, "기수 수정에 성공했습니다.");

    private final int code;
    private final HttpStatus status;
    private final String message;

    UserResponseCode(int code, HttpStatus status, String message) {
        this.code = code;
        this.status = status;
        this.message = message;
    }
}

File: src/main/kotlin/com/weeth/domain/user/presentation/UserResponseCode.kt
package com.weeth.domain.user.presentation

import com.weeth.global.common.response.ResponseCodeInterface
import org.springframework.http.HttpStatus

enum class UserResponseCode(
    override val code: Int,
    override val status: HttpStatus,
    override val message: String,
) : ResponseCodeInterface {
    USER_UPDATE_SUCCESS(10901, HttpStatus.OK, "회원 정보가 성공적으로 수정되었습니다."),
    JWT_REFRESH_SUCCESS(10902, HttpStatus.OK, "토큰 재발급에 성공했습니다."),
    SOCIAL_LOGIN_SUCCESS(10903, HttpStatus.OK, "소셜 로그인이 성공적으로 처리되었습니다."),
    USER_TERMS_AGREE_SUCCESS(10904, HttpStatus.OK, "약관 동의가 성공적으로 처리되었습니다."),
    INQUIRY_SEND_SUCCESS(10905, HttpStatus.OK, "문의가 성공적으로 접수되었습니다."),
    USER_LEFT_SUCCESS(10906, HttpStatus.OK, "위드 탈퇴가 완료되었습니다."),
    USER_PROFILE_CREATED_SUCCESS(10907, HttpStatus.CREATED, "프로필이 성공적으로 생성되었습니다."),
    USER_PROFILE_FIND_ALL_SUCCESS(10908, HttpStatus.OK, "프로필 목록을 성공적으로 조회했습니다."),
    USER_PROFILE_FIND_SUCCESS(10909, HttpStatus.OK, "프로필을 성공적으로 조회했습니다."),
    USER_PROFILE_UPDATED_SUCCESS(10910, HttpStatus.OK, "프로필이 성공적으로 수정되었습니다."),
    USER_PROFILE_ASSIGNMENT_UPDATED_SUCCESS(10911, HttpStatus.OK, "동아리별 사용 프로필이 성공적으로 변경되었습니다."),
    USER_PROFILE_DELETED_SUCCESS(10912, HttpStatus.OK, "프로필이 성공적으로 삭제되었습니다."),
    USER_MY_PAGE_FIND_SUCCESS(10913, HttpStatus.OK, "마이페이지 정보를 성공적으로 조회했습니다."),
    USER_MY_POSTS_FIND_SUCCESS(10914, HttpStatus.OK, "내가 쓴 글을 성공적으로 조회했습니다."),
    USER_ATTENDED_SESSIONS_FIND_SUCCESS(10915, HttpStatus.OK, "출석한 세션을 성공적으로 조회했습니다."),
    USER_PROFILE_IMAGE_DELETED_SUCCESS(10916, HttpStatus.OK, "프로필 사진이 성공적으로 삭제되었습니다."),
    USER_PROFILE_HEADER_IMAGE_DELETED_SUCCESS(10917, HttpStatus.OK, "프로필 헤더 사진이 성공적으로 삭제되었습니다."),
    USER_PROFILE_ASSIGNABLE_CLUBS_FIND_SUCCESS(10918, HttpStatus.OK, "프로필을 사용할 수 있는 동아리 목록을 성공적으로 조회했습니다."),
    USER_MY_PENALTIES_FIND_SUCCESS(10919, HttpStatus.OK, "페널티 목록을 성공적으로 조회했습니다."),
    USER_PENALTY_RULE_FIND_SUCCESS(10920, HttpStatus.OK, "패널티 규정을 성공적으로 조회했습니다."),
}
